rust/clients/host_agent/src/hostd/inventory.rs

```diff
@@ -26,75 +28,106 @@ pub fn should_check_inventory(
     now.signed_duration_since(start) > check_interval_duration
 }
 
+/// Periodically checks for inventory changes and publishes them on a NATS subject.
+/// Never returns.
 pub async fn run(
     host_client: JsClient,
     host_id: &str,
-    inventory_file_path: &str,
+    inventory_file_path_str: &str,
     host_inventory_check_interval_sec: u64,
-    starting_inventory: HoloInventory,
-) -> Result<(), ServiceError> {
+) -> ! {
     log::info!("Host Agent Client: starting Inventory job...");
 
-    // Store latest inventory record in memory
-    starting_inventory
-        .save_to_file(inventory_file_path)
-        .map_err(|e| {
-            ServiceError::internal(
-                e.to_string(),
-                Some("Failed to save host inventory to file.".to_string()),
-            )
-        })?;
-
     let one_hour_interval = tokio::time::Duration::from_secs(host_inventory_check_interval_sec);
```

Comment:
```suggestion
    let interval = tokio::time::Duration::from_secs(host_inventory_check_interval_sec);
```
I do prefer the generic naming applied in the pr 100/120.